src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -68,38 +97,112 @@ public CompletionHandler(ILspSolutionProvider solutionProvider) : base(solutionP
 
             return new LSP.VSCompletionList
             {
-                Items = list.Items.Select(item => CreateLSPCompletionItem(request, item, lspVSClientCapability)).ToArray(),
+                Items = list.Items.Select(item => CreateLSPCompletionItem(request, item, lspVSClientCapability, completionTrigger)).ToArray(),
                 SuggesstionMode = list.SuggestionModeItem != null,
             };
 
-            // local functions
-            static LSP.CompletionItem CreateLSPCompletionItem(LSP.CompletionParams request, CompletionItem item, bool useVSCompletionItem)
+            // Local functions
+            bool IsValidTriggerCharacterForDocument(Document document, string triggerCharacter)
+            {
+                if (document.Project.Language == LanguageNames.CSharp)
+                {
+                    return _csTriggerCharacters.Contains(triggerCharacter);
+                }
+                else if (document.Project.Language == LanguageNames.VisualBasic)
+                {
+                    return _vbTriggerCharacters.Contains(triggerCharacter);
+                }
+
+                return true;
```

Comment:
hrmm

Reply:
added comment explaining the behavior (this is from the reverted PR)